Docker compose

elainearbaugh opened the pull request:
Update cabot to use docker-compose and fix the tests

elainearbaugh commented:
(WIP)

jimjh commented on setup.py:
```diff
@@ -1,7 +1,14 @@
+with open(requirements_file) as f:
+    requirements = [line.rstrip('\n')
+                    for line in f
+                    if line and not line.startswith('#')]
```
Have you considered using `pkg_resources.parse_requirements`?

e.g. https://github.com/Affirm/datatools/blob/master/dbimport/setup.py#L27

jimjh commented on docker-entrypoint.sh:
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+set -e
```
I also recommend

```
set -o pipefail
set -o nounset
```

See [Coding Guidelines](https://github.com/Affirm/salt/blob/develop/CODING_GUIDELINES.md).

jimjh commented on cabot/cabotapp/influx.py:
```diff
@@ -35,7 +35,7 @@ def get_data(pattern, selector='value',
              fetchall=False):
     '''
-    Query a metric and it's data from influxdb.
+    Query a metric and its data from influxdb.
```
🖖

jimjh commented on Dockerfile:
```diff
@@ -0,0 +1,35 @@
+        libsasl2-dev \
+        libldap2-dev \
+        libpq-dev \
+        npm
```
I believe you need

```
DEBIAN_FRONTEND=noninteractive \
apt-get -qqy update && \
apt-get -qqy install python-dev ...
```

elainearbaugh replied:
added env debian_frontend

elainearbaugh commented:
@Affirm/infrastructure please review?
- Copied docker-compose code from arachnys/cabot
- Fixed tests that were failing
- Added test for jenkins success since all the others were testing for failure

gregsterin commented on cabot/cabotapp/monitor.py:
```diff
@@ -28,14 +28,16 @@
+        DIMENSIONS = {
+            'instance-id': _instance['instance-id'],
+        }
+        DIMENSIONS = {}
```
This wipes the dimensions we just set, no?

elainearbaugh replied:
oops thanks for catching

elainearbaugh commented on setup.py:
```diff
@@ -1,6 +1,12 @@
+    requirements = pkg_resources.parse_requirements(f)
 
 # pull in active plugins
 plugins = env['CABOT_PLUGINS_ENABLED'].split(',') if 'CABOT_PLUGINS_ENABLED' in env else ["cabot_alert_hipchat", "cabot_alert_twilio", "cabot_alert_email"]
```
Yeah not sure why they had CABOT_PLUGINS_ENABLED and requirements-pluginx.txt...
sorry which package?

elainearbaugh replied:
Guess it's not needed for the setup with docker-compose? I copied this part from arachnys

We don't use travis-ci anyways right? so maybe I should just get rid travis.yml...

srikiraju replied:
I don't really have any context about this, I just thought that change looked strange. I can offer a 🌮

elainearbaugh replied:
I got rid of it 🌯

elainearbaugh commented:
merged with https://github.com/Affirm/cabot/pull/6
